linera-base/src/crypto/mod.rs

```diff
@@ -213,6 +229,58 @@ impl TryFrom<&[u8]> for AccountSignature {
     }
 }
 
+/// A trait for signing keys.
+#[cfg_attr(not(web), async_trait)]
+#[cfg_attr(web, async_trait(?Send))]
+pub trait SigningKey {
+    /// The public key type corresponding to this signing key.
+    type PublicKey: Into<AccountPublicKey> + std::fmt::Debug;
```

Comment:
I think this is an antipattern: the things that are `Into<AccountPublicKey>` are roughly the contents of the `AccountPublicKey` enum, so this associated type doesn't add much information over just returning an `AccountPublicKey` (as I see you have, indeed!).

Reply:
No I think using `AccountPublicKey` is exactly right.  I think the associated `PublicKey` type is what isn't necessary, and it makes it harder to make a trait object in the `ClientContext` later on.